README.md

```diff
@@ -4,31 +4,95 @@ The `ray_julia_jll` is the Julia C++ wrapper that interfaces with the Ray.io pro
+
+#### MacOS
+
+Based on [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#preparing-to-build-ray-on-macos):
+
+```sh
+brew update
+brew install wget
+brew install bazelisk
+```
+
+#### Linux
+
+Based on [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#preparing-to-build-ray-on-linux), assuming Ubuntu:
+
+```sh
+# Add a PPA containing gcc-9 for older versions of Ubuntu.
+sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+sudo apt-get update
+sudo apt-get install -y build-essential curl gcc-9 g++-9 pkg-config psmisc unzip
+sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90 \
+              --slave /usr/bin/g++ g++ /usr/bin/g++-9 \
+              --slave /usr/bin/gcov gcov /usr/bin/gcov-9
+```
+
+Install [Bazelisk](https://github.com/bazelbuild/bazelisk#readme) by downloading the latest binary for your system from the [releases page](https://github.com/bazelbuild/bazelisk/releases) somewhere on your `$PATH` (like `$HOME/bin/`).  For instance, as of writing this was the latest release for an AMD64 system:
+
+```sh
+curl -Lo "$HOME/bin/bazel" "https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-linux-amd64"
+chmod +x "$HOME/bin/bazel"
+```
+
+Alternatively, if you have `npm` installed you can install Bazelisk that way:
+
+```sh
+npm install -g @bazel/bazelisk
+```
+
+### Prepare python environment
+
+We recommend always using the same virtualenv as Bazel can cause unnecessary rebuilds when using different versions of Python.
+
+```sh
+python -m venv venv
+source venv/bin/activate
+# make sure we're using up-to-date version of pip and wheel:
+python -m pip install --upgrade pip wheel
+```
+
+(These are the same as the [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#prepare-the-python-environment) but copied here for consistency.)
+
+### Build wrapper
 
 ```sh
-# We recommend always using the same virtualenv as Bazel can cause unnecessary rebuilds when
-# using different versions of Python.
 python3 -m venv venv
 source venv/bin/activate
 
 # Build the required libraries
 julia --project=ray_julia_jll -e 'using Pkg; Pkg.build(verbose=true)'
+```
+
+### Build Ray CLI/server
 
+```sh
 # Build the Ray CLI. Based upon these instructions:
 # https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#building-ray-on-linux-macos-full
 python -m pip install --upgrade pip wheel
 cd ray_julia_jll/deps/ray/python
 pip install -e . --verbose
 cd -
-
-# Run the tests
-julia --project -e 'using Pkg; Pkg.test()'
 ```
 
-## Sources
-
-The tarballs for `ray_julia_jll.jl` have been built from these sources:
+### Validate installation
 
-* ray v2.5.1
-* libcxxwrap_julia
+```sh
+# Run the tests
+julia --project -e 'using Pkg; Pkg.develop(; path="./ray_julia_jll"); Pkg.test()'
```

Comment:
is a side-effect of this to get `Ray.jl` to `dev` `ray_julia_jll`?

If so might be good to call that out? otherwise I'd wonder why we don't just do `--project=ray_julia_jll`

Reply:
The Ray.jl tests are at this point a bit more extensive.  Dev'ing the jll is necessary since otherwise it tries to `add` all the dependencies when instantiating the test environment; if there's a manifest alreayd present it'll re-use taht to the extent that it can.

I do think it's worth calling it out (that you currently need to `dev` _both_ the Ray.jl and jll)